src/lightgbm/src/main/scala/LightGBMClassifier.scala

```diff
@@ -78,10 +78,13 @@ class LightGBMClassifier(override val uid: String)
       getIsUnbalance, getVerbosity, categoricalIndexes, classes, metric, getBoostFromAverage, getBoostingType)
     log.info(s"LightGBMClassifier parameters: ${trainParams.toString}")
     val networkParams = NetworkParams(getDefaultListenPort, inetAddress, port)
-
+    val validationData =
+      if (get(validationIndicatorCol).isDefined && get(validationIndicatorCol).get.nonEmpty)
```

Comment:
encode the nonempty check in param validator

Reply:
it can be empty, we just don't use the validation column.  but I added a non-null check.

Reply:
actually, I think it is fine if it is null, I modified this part of the code a bit differently anyway